chore: Chromium 67

alexeykuzmin opened the pull request:
Project Board https://github.com/electron/electron/projects/11
PR to Electron https://github.com/electron/electron/pull/13756

alexeykuzmin committed:
Use Chromium 67.0.3396.99

alexeykuzmin committed:
Update chromium/dcheck.patch

Trivial patch fixes.

alexeykuzmin committed:
Update chromium/dcheck.patch

Create separate files for RWHVMac and RWHVCocoa
https://chromium-review.googlesource.com/969404

alexeykuzmin committed:
Update chromium/dcheck.patch

The Great Blink mv for source files

alexeykuzmin committed:
Remove Chromium backports that are already available in 67.0.3396.99

alexeykuzmin committed:
Remove V8 backports that are already available in 6.7.288.46

alexeykuzmin committed:
Remove webrtc patches that are already available in Chromium 67.0.3396.99

alexeykuzmin committed:
Update common/chromium/accelerator.patch

alexeykuzmin committed:
Update common/chromium/blink_file_path.patch

The Great Blink mv for source files

alexeykuzmin committed:
Update common/chromium/blink_local_frame.patch

The Great Blink mv for source files

alexeykuzmin committed:
Update common/chromium/blink_world_context.patch

The Great Blink mv for source files

alexeykuzmin committed:
Update common/chromium/can_create_window.patch

alexeykuzmin committed:
Update common/chromium/compositor_delegate.patch

alexeykuzmin committed:
Update common/chromium/latency_info.patch

alexeykuzmin committed:
Update common/v8/array_buffer.patch

alexeykuzmin committed:
Update common/v8/export_platform.patch

alexeykuzmin committed:
Update common/v8/backport_39d546a.patch

alexeykuzmin committed:
Update common/v8/backport_ff0a97933.patch

alexeykuzmin committed:
Remove common/v8/backport_b49206d.patch

The change has been backported to the 6.7 branch.
See https://chromium-review.googlesource.com/1058046.

alexeykuzmin committed:
Update chromiumcontent/BUILD.gn for Chromium 67

media: Remove widevinecdmadapter target
https://chromium-review.googlesource.com/957868

Rename buildflag_headers for consistency
https://chromium-review.googlesource.com/975990

The Great Blink mv for source files, part 2.
https://chromium-review.googlesource.com/1001153

alexeykuzmin committed:
Update common/chromium/can_create_window.patch

Move file, file_path and file_info from common to base
https://chromium-review.googlesource.com/959423

Move time.mojom from common to base
https://chromium-review.googlesource.com/962872

alexeykuzmin committed:
Update common/chromium/render_widget_host_view_base.patch

alexeykuzmin committed:
Update common/chromium/render_widget_host_view_mac.patch

Create separate files for RWHVMac and RWHVCocoa
https://chromium-review.googlesource.com/969404

Remove redundant code from RWHVCocoa, move some code to RWHVNSViewBridge
https://chromium-review.googlesource.com/977227

alexeykuzmin committed:
Update common/chromium/webgl_context_attributes.patch

The Great Blink mv for source files, part 2.
https://chromium-review.googlesource.com/1001153

alexeykuzmin committed:
Update common/chromium/worker_context_will_destroy.patch

The Great Blink mv for source files, part 2.
https://chromium-review.googlesource.com/1001153

alexeykuzmin committed:
Update common/chromium/export_blink_webdisplayitemlist.patch

The Great Blink mv for source files, part 2.
https://chromium-review.googlesource.com/1001153

alexeykuzmin committed:
Update common/chromium/disable_user_gesture_requirement_for_beforeunload_dialogs.patc

The Great Blink mv for source files, part 2
https://chromium-review.googlesource.com/1001153

alexeykuzmin committed:
Update common/chromium/gin_enable_disable_v8_platform.patch

alexeykuzmin committed:
Update common/chromium/blink-worker-enable-csp-in-file-scheme.patch

Worker: Rename WorkerScriptLoader to WorkerClassicScriptLoader
https://chromium-review.googlesource.com/965662

The Great Blink mv for source files, part 2.
https://chromium-review.googlesource.com/1001153

alexeykuzmin committed:
Remove exclude_next_version_mini_installer_from_deps.patch

alexeykuzmin committed:
FIXME: Do not apply crashpad-disabled-windows.patch

alexeykuzmin committed:
Update common/chromium/blink_fix_prototype_assert.patch

The Great Blink mv for source files, part 2.
https://chromium-review.googlesource.com/1001153

alexeykuzmin committed:
Update common/chromium/pepper_flash.patch

See "Remove output protection pepper APIs"
https://chromium-review.googlesource.com/988254

and "Remove platform verification pepper APIs"
https://chromium-review.googlesource.com/1007885

alexeykuzmin committed:
Update common/chromium/sysroot.patch

Remove the Debian stretch sysroot
https://chromium-review.googlesource.com/950310

alexeykuzmin committed:
Update common/chromium/disable_scroll_begin_dcheck.patch

Reland Touchscreen browser side fling and its follow up cls.
https://chromium-review.googlesource.com/901959

alexeykuzmin committed:
Update common/ffmpeg/build_gn.patch

alexeykuzmin committed:
Update common/webrtc/webrtc-rwlock_null.patch

Move rw_lock_wrapper.h to rtc_base/synchronization/
https://webrtc-review.googlesource.com/21328

alexeykuzmin committed:
Update common/webrtc/webrtc-desktop_capturer_mac.patch

alexeykuzmin committed:
Update common/chromium/render_widget_host_view_mac.patch

Introduce interface between RWHVMac and RWHVCocoa
https://chromium-review.googlesource.com/969975

alexeykuzmin committed:
Update common/chromium/can_create_window.patch

Create a TYPE_CHUNKED_DATA_PIPE DataElement.
https://chromium-review.googlesource.com/908828

alexeykuzmin committed:
Update Electron build configs for Chromium 67

There are errors about some generated headers not being available.
It probably happens because of Jumbo builds.

The Great Blink mv for source files, part 2
https://chromium-review.googlesource.com/1001153

Removes web/ lib and dependencies from Blink.
https://chromium-review.googlesource.com/602093

Remove web/ finally!!!
https://chromium-review.googlesource.com/606812

Move //components/filesystem to //components/services/
https://chromium-review.googlesource.com/c/chromium/src/+/948913

alexeykuzmin committed:
Update common/chromium/blink_no_private_api.patch

The Great Blink mv for source files, part 2
https://chromium-review.googlesource.com/1001153

alexeykuzmin committed:
Update common/chromium/no_private_api.patch

macviews: fix menu transparency, remove some 10.9 code
https://chromium-review.googlesource.com/987476

macviews: materialize macviews menus
https://chromium-review.googlesource.com/998016

alexeykuzmin committed:
Disable widevine on Linux

media: Add assert for "enable_widevine" gn arg
https://chromium-review.googlesource.com/957707

alexeykuzmin committed:
Use clang on Windows

alexeykuzmin commented on patches/common/chromium/.patches.yaml:
```diff
@@ -311,3 +311,9 @@ patches:
+  file: string_piece_check_disable.patch
+  description: |
+    Disables some static checks on constexpr's in the string_piece helper to
+    avoid incompatibilities with node's CHECK macro
```
@MarshallOfSound 
Please preserve newlines at the end of files.

alexeykuzmin commented on patches/common/chromium/string_piece_check_disable.patch:
```diff
@@ -0,0 +1,28 @@
+
+   constexpr void remove_suffix(size_type n) {
+-    CHECK(n <= length_);
++    // CHECK(n <= length_);
```
@MarshallOfSound Looks scary.
https://github.com/electron/electron/projects/11#card-11512594

deepak1556 replied:
The patch is not needed, the fix is on electron side.

MarshallOfSound replied:
Heh, heavy-handed approach 😆

alexeykuzmin commented:
@MarshallOfSound @deepak1556 
I'll clean up the branch later this week to remove all redundant commits.

MarshallOfSound commented on chromiumcontent/build_libs.py:
```diff
@@ -143,11 +143,12 @@ def gen_list(out, name, obj_dirs, exclude_dirs=[]):
         out,
         "obj_components",
         [
-            "components/appdu",
```
Oh my.... I spent like a solid hour this afternoon wondering why this didn't work.... 😢

alexeykuzmin committed:
Do not collect static libraries PDB files for dist on Windows

Starting Chromium 67 we should move to libcc+Electron combined build
so create-dist shouldn't be needed anymore anyway.

nornagon commented on chromiumcontent/args/ffmpeg.gn:
```diff
@@ -3,11 +3,15 @@ is_component_build = false
+# As of Chromium 67.0.3396.99 widevine is only supported on Mac and Windows.
+if (target_os != "linux") {
+  enable_widevine = true
+}
```
This seems like a bummer; is upstream no longer supporting this?

deepak1556 replied:
We don't have the chrome component updater setup in electron, also for proprietary reasons we ask users to get their own widevine libraries, electron only integrates mechanism required to load cdm. So this build flag will not affect us directly.

alexeykuzmin replied:
@nornagon It is not supported on Linux in Ch67. I have no idea if there are plans to support it in the future.

nornagon committed:
fix: backport legacy OCSP APIs in boringssl (#628)

One of Electron's tests exposed the fact
that boringssl didn't support the OCSP stapling APIs that nodejs was calling.
This backports a patch from boringssl that adds the APIs
that nodejs expects, and causes the test to pass.

codebytere committed:
src: add patches for Node upgrade v10.6.0 (#626)

Update patches in preparation for the upgrade to Chromium 67

Included commits:
62ca2cf21c deps: cherry-pick 70c4340 from upstream V8
ab27e0e785 deps: cherry-pick acc336c from upstream V8
37a5c8c2ff deps: cherry-pick b20faff from upstream V8
4663d1c22e deps: backport aa6ce3e from upstream V8
5d7218965d deps: cherry-pick 5dd3395 from upstream V8
14bb905d18 deps: V8: cherry-pick a440efb27f from upstream
35d6661973 deps: cherry-pick 6989b3f6d7 from V8 upstream
4e788dc2f5 deps: backport 91ddb65d from upstream V8

alexeykuzmin committed:
Update common/chromium/dcheck.patch

alexeykuzmin committed:
Update common/chromium/windows_cc_wrapper.patch

alexeykuzmin committed:
Update common/chromium/enable_osr_components.patch

alexeykuzmin committed:
Update common/v8/.patches.yaml format

alexeykuzmin committed:
build: turn enable_widevine flag on by default on Mac and Windows

codebytere committed:
feat: implement ArrayBuffer realloc (#669)

* feat: implement ArrayBuffer realloc

alexeykuzmin committed:
build: remove common/chromium/vmware_gpu_acceleration.patch

alexeykuzmin committed:
fix: implement SSL_get_tlsext_status_type() in BoringSSL

For Node.js 10.6.0.

deepak1556 reviewed (approved):
LGTM

alexeykuzmin committed:
Update common/chromium/blink_initialization_order.patch

https://chromium-review.googlesource.com/1030530

nornagon committed:
fix: [m67] patch leveldb to not redefine ssize_t (#682)
